### PR TITLE
Spanish translation for Lesson 2 Chapters 9 to 11

### PR DESCRIPTION
**Pull request by h-nick**
Added Spanish translations for:
 Chapter 9: Assert
 Chapter 10: Calling a Contract
 Chapter 11: Putting it Together

### CODE REVIEW

**Comment by vasa-develop** [comment]
@gzanitti Can you review this one?

File: es/2/assert.md [hunk 1]
```diff
@@ -1,5 +1,67 @@
-<!-- Add translation for the following page: https://vyper.fun/#/2/assert
-Do NOT change the code below. The below code runs the code editor -->
+# Capítulo 9: Afirmaciones
+
+En el capítulo anterior, agregamos la interfaz `WildPokemons` al contrato del entrenador. En los siguientes capítulos, empezaremos a escribir las funciones para pelear con los pokemones salvajes usando la función `battle` de la interfaz `WildPokemon`.
+
+En el contrato actual, cuando llamamos la función `createTrainer`, un nuevo entrenador es creado con un solo pokemon.
```
Comment (gzanitti) [inline es/2/assert.md]:
```suggestion
En el contrato actual, cuando llamamos a la función `createTrainer`, un nuevo entrenador es creado con un solo pokemon.
```

File: es/2/assert.md [hunk 2]
```diff
@@ -1,5 +1,67 @@
...
+
+Adicionalmente puedes agregar un mensaje de error _opcional_.
+
+```vyper
+@external
+def applyForDriversLicense(String[32]: name, uint256: age):
+    assert age < 18 , "no es elegible para una licencia de conducir"
+```
+
+Aquí, si el parámetro `age` que recibe la función `applyForDriversLicense` is menor a `18`, entonces la transacción es revertida con el mensaje `"no es elegible para una licencia de conducir".
```
Comment (gzanitti) [inline es/2/assert.md]:
```suggestion
Aquí, si el parámetro `age` que recibe la función `applyForDriversLicense` es menor a `18`, entonces la transacción es revertida con el mensaje `"no es elegible para una licencia de conducir".
```

File: es/2/assert.md [hunk 3]
```diff
@@ -1,5 +1,67 @@
...
+trainerToPokemon: HashMap[address, HashMap[uint256, Pokemon]]
+```
+
+Para seleccionar un `Pokemon` específico de un entrenador, necesitamos dos cosas:
+
+- `address` del entrenador ejecutando el contrato.
+
+- Un índice `uint256` que seleccionará un pokemon específico del entrenador.
+
+Recordemos del [Capítulo 2](https://vyper.fun/#/2/msg-sender) que puedes obtener la dirección del ejecutor del contrato usando `msg.sender`.
```
Comment (gzanitti) [inline es/2/assert.md:41]:
```suggestion
Recordemos del [Capítulo 2](https://vyper.fun/#/2/msg-sender) que puedes obtener la dirección de quien está ejecutando el contrato usando `msg.sender`.
```

File: es/2/assert.md [hunk 4]
```diff
@@ -1,5 +1,67 @@
...
+
+Cuando un nuevo entrenador es creado, el valor de `self.trainerPokemonCount[msg.sender]` para el entrenador es `0`.
+
+El primer pokemon es agregado al índice `0` del mapeo `trainerToPokemon`.
+
+Mientras vamos incrementando `trainerPokemonCount`, subsecuentes pokemones son añadidos a índices superiores.
+
+¿Qué ocurre si un entrenador con un solo pokemon ejecuta la función `battleWildPokemon` con un `pokemonIndex` mayor a `0`?
+
+Como no hay pokemones asociados con `pokemonIndex` mayor a `0` para este entrenador, la transacción causaría un error durante el tiempo de ejecución.
```
Comment (gzanitti) [inline es/2/assert.md]:
```suggestion
Como no hay pokemones asociados con `pokemonIndex` mayor a `0` para este entrenador, la transacción causaría un error en tiempo de ejecución.
```

File: es/2/assert.md [hunk 5]
```diff
@@ -1,5 +1,67 @@
...
+
+2. Para mitigar esta situación, dentro de la función `battleWildPokemon`, agregamos una declaración `assert` que chequea si `pokemonIndex` es menor que `self.trainerPokemonCount[msg.sender]`. Si la condición es `Falsa`, retornamos el mensaje de error: `"Índice No Válido Provisto"`.
```
Comment (gzanitti) [inline es/2/assert.md]:
```suggestion
2. Para mitigar esta situación, dentro de la función `battleWildPokemon`, agregamos una declaración `assert` que chequea si `pokemonIndex` es menor que `self.trainerPokemonCount[msg.sender]`. Si la condición es `False`, retornamos el mensaje de error: `"Índice No Válido Provisto"`.
```

File: es/2/calling_a_contract.md [hunk 6]
```diff
@@ -1,5 +1,62 @@
...
+- `battleResult`: `bool`
+- `newPokemonName`: `String[32]`
+- `newPokemonDNA`: `uint256`
+- `newPokemonHP`: `uint256`
+
+1. Dentro de `battleWildPokemon`, define estas 4 variables de función e inicialízalas usando la función incorporada `empty`.
+
+2. Define una variable de almacenamiento con `constant` `address` llamada `WILD_POKEMON` con el valor `0xC84a08B45CF0FC28EFC8caE8B7Fc1d062115048e`.
+
+3. Dentro de `battleWildPokemon`, usa `WildPokemons` y `WILD_POKEMON` para llamar la función `battle` que recibe un solo parámetro de tipo `Pokemon`. Este parámetro puede ser obtenido usando el mapeo `trainerToPokemon` enviando las 2 llaves: `msg.sender` como la primera y `pokemonIndex` como la segunda.
```
Comment (gzanitti) [inline es/2/calling_a_contract.md]:
```suggestion
3. Dentro de `battleWildPokemon`, usa `WildPokemons` y `WILD_POKEMON` para llamar a la función `battle` que recibe un solo parámetro de tipo `Pokemon`. Este parámetro puede ser obtenido usando el mapeo `trainerToPokemon` enviando 2 claves: `msg.sender` como la primera y `pokemonIndex` como la segunda.
```

File: es/2/putting_it_together.md [hunk 7]
```diff
@@ -1,5 +1,38 @@
-<!-- Add translation for the following page: https://vyper.fun/#/2/putting_it_together
-Do NOT change the code below. The below code runs the code editor -->
+# Capítulo 11: Juntándolo todo
+
+En el capítulo anterior, llamamos la función `battle` perteneciente al contrato del pokemon salvaje, desde el contrato del entrenador.
```
Comment (gzanitti) [inline es/2/putting_it_together.md]:
```suggestion
En el capítulo anterior, llamamos a la función `battle` perteneciente al contrato del pokemon salvaje, desde el contrato del entrenador.
```

File: es/2/putting_it_together.md [hunk 8]
```diff
@@ -1,5 +1,38 @@
...
+
+## Ponlo a prueba
+
+Como discutimos, tenemos 2 condiciones basadas en la variable `battleResult`. Si `battleResult` es `True` eso significa que el pokemon del entrenador ha ganado, de lo contrario (si es `False`), entonces el pokemon del entrenador perdió.
+
+1. Crea una expresión if/else con la condición `battleResult`.
+
+2. Si `battleResult` es `True`:
+
+  - Incrementa el número de `matches` para el pokemon del entrenador por `1` (utiliza el operador aritmético `+=` para mantener el código limpio). Para acceder al pokemon del entrenador, utiliza el mapeo `trainerToPokemon` con la primera llave `msg.sender` y la segunda llave `pokemonIndex`.
```
Comment (gzanitti) [inline es/2/putting_it_together.md:17]:
```suggestion
  - Incrementa el número de `matches` para el pokemon del entrenador en `1` (utiliza el operador aritmético `+=` para mantener el código limpio). Para acceder al pokemon del entrenador, utiliza el mapeo `trainerToPokemon` con la primera clave `msg.sender` y la segunda clave `pokemonIndex`.
```

File: es/2/putting_it_together.md [hunk 9]
```diff
@@ -1,5 +1,38 @@
...
+  - Incrementa el número de `wins` para el pokemon del entrenador por `1` (utiliza el operador aritmético `+=` para mantener el código limpio).
```
Comment (gzanitti) [inline es/2/putting_it_together.md:18]:
```suggestion
  - Incrementa el número de `wins` para el pokemon del entrenador en `1` (utiliza el operador aritmético `+=` para mantener el código limpio).
```

File: es/2/putting_it_together.md [hunk 10]
```diff
@@ -1,5 +1,38 @@
...
+  - Crea una nueva variable llamada `newPokemon` de tipo `Pokemon` y asigna su valor a un nuevo Pokemon creado usando la estructura `Pokemon` y los siguientes parámetros:
+    - name: `newPokemonName`
+    - dna: `newPokemonDNA`
+    - HP: `newPokemonHP`
+    - matches: `1` (ya que el pokemon salvaje acaba de pelear)
+    - wins: `0`
+  - Agrega este `newPokemon` al mapeo `pokemonList` usando la llave `totalPokemonCount`.
+  - Incrementa `totalPokemonCount` por `1`.
```
Comment (gzanitti) [inline es/2/putting_it_together.md:26]:
```suggestion
  - Incrementa `totalPokemonCount` en `1`.
```

File: es/2/putting_it_together.md [hunk 11]
```diff
@@ -1,5 +1,38 @@
...
+  - Agrega el `newPokemon` al mapeo `trainerToPokemon` usando como primera llave `msg.sender` y como segunda llave `trainerPokemonCount[msg.sender]`.
```
Comment (gzanitti) [inline es/2/putting_it_together.md]:
```suggestion
  - Agrega el `newPokemon` al mapeo `trainerToPokemon` usando como primera clave `msg.sender` y como segunda clave `trainerPokemonCount[msg.sender]`.
```

File: es/2/putting_it_together.md [hunk 12]
```diff
@@ -1,5 +1,38 @@
...
+  - Incrementa `trainerPokemonCount[msg.sender]` por `1` usando el operador aritmético `+=`.
+  - Emite el evento `NewPokemonCreated` con los siguientes parámetros: `newPokemonName`, `newPokemonDNA` y `newPokemonHP`.
+
+3. Si `battleResult` es `False`, incrementa el número de `matches` para el pokemon del entrenador por `1` (utiliza el operador aritmético `+=` para mantener el código limpio).
```
Comment (gzanitti) [inline es/2/putting_it_together.md:31]:
```suggestion
3. Si `battleResult` es `False`, incrementa el número de `matches` para el pokemon del entrenador en `1` (utiliza el operador aritmético `+=` para mantener el código limpio).
```

**Review by gzanitti — CHANGES_REQUESTED** [review]

**Comment by gzanitti** [comment]
> @gzanitti Can you review this one?

@vasa-develop Done. Take a look and let's merge if you agree :)

**Comment by gzanitti** [comment]
Congrats @hniks! 
The PR has been merged.
Thanks for your help 😃

**Pull request merged**